### PR TITLE
BE: Notifications model and routes

### LINKED ISSUE #4: BE: Notifications
Make notifications model. Think about what types of notifications we might want to show the user.
- type
- title
- description
- read/unread
- date
Routes:
- Create notification
- Mark notification as read
- Get all notifications
- Get unread notifications

### PR DESCRIPTION
**Pull request by Habib-Ahmad**
### What this PR does (required):
- The notification model was created as well as its routes.
- The routes are create new notification, mark notification as read, get all notifications and get all unread notifications.
- closes #4 

### Any information needed to test this feature (required):
- Routes can be tested using any API platform such as Postman. Just add the bearer token from the login Endpoint.

### CODE REVIEW

File: server/controllers/notifications.js [hunk 1]
```diff
@@ -0,0 +1,93 @@
+const Notification = require("../models/Notifications");
+const asyncHandler = require("express-async-handler");
+
+// @route POST /notification/create
+// @desc create new notification
+// @access Private
```
Comment (sundayezeilo) [inline server/controllers/notifications.js:16]: Good documentation

File: server/controllers/notifications.js [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+exports.createNotification = asyncHandler(async (req, res, next) => {
+  const { title, type, description, read, date } = req.body;
```
Comment (sundayezeilo) [inline server/controllers/notifications.js]: This is a good security practice :+1:

File: server/controllers/notifications.js [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+    });
+  } else {
+    res.status(400);
+    throw new Error("Invalid notification data");
+  }
+});
+
+// @route PUT notifications/markAsRead
+// @desc mark notification as read
+// @access Private
```
Comment (sundayezeilo) [inline server/controllers/notifications.js:64]: :+1:

File: server/controllers/notifications.js [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+  res.status(200).json({
+    success: {
+      notifications: notification,
+    },
+  });
+});
+
+// @route GET /notifications/getAll
+// @desc Get all user notifications
+// @access Private
```
Comment (sundayezeilo) [inline server/controllers/notifications.js:83]: :+1:

File: server/controllers/notifications.js [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+  res.status(200).json({
+    success: {
+      notifications: notifications,
+    },
+  });
+});
+
+// @route GET /notifications/getUnread
+// @desc Get all unread user notifications
+// @access Private
```
Comment (sundayezeilo) [inline server/controllers/notifications.js:101]: :+1:

**Review by sundayezeilo — COMMENTED** [review]
LGTM :smiley: But please don't forget to link your ticket next time.

**Comment by cl-codes** [comment]
Nice work Ahmad - I tested the four new routes in Postman and they all worked well!  😊

**Review by moffatethan — APPROVED** [review]

**Pull request merged**